File: android/app/src/main/java/io/nexy/android/ui/voice/VoiceTranscript.kt
package io.nexy.android.ui.voice

private val EMPTY_AUDIO_MARKER = Regex(
    pattern = """^\[\s*blank[\s_-]*audio\s*]$""",
    option = RegexOption.IGNORE_CASE,
)

/**
 * Returns editable speech text, or null when the recognizer reports no speech.
 *
 * whisper.cpp can return `[BLANK_AUDIO]` as a successful transcription for an
 * empty or silent recording. That marker must never be inserted in a composer.
 */
internal fun usableVoiceTranscript(text: String): String? =
    text.trim().takeIf { it.isNotEmpty() && !EMPTY_AUDIO_MARKER.matches(it) }

File: src/renderer/lib/voice-transcript.ts
const EMPTY_AUDIO_MARKER = /^\[\s*blank[\s_-]*audio\s*\]$/i

/**
 * Returns editable speech text, or null when the recognizer reports no speech.
 *
 * whisper.cpp can return `[BLANK_AUDIO]` as a successful transcription for an
 * empty/silent recording. That marker is transport metadata, not user text.
 */
export function usableVoiceTranscript(text: string): string | null {
  const trimmed = text.trim()
  if (!trimmed || EMPTY_AUDIO_MARKER.test(trimmed)) return null
  return trimmed
}
